### ISSUE TITLE
Error when trying to set memory amount in Python

### ISSUE THREAD

**Issue by benmullins** [clab/dynet#359]
Hi,

I'm trying to change the amount of memory that DyNet uses from within Python code, using the interface recently added (where you instantiate a `DyParams` object and then set parameters with it). I can create the `DyParams` object fine, but when I try to use it to set the memory, this happens:

```
>>> dyparams.set_mem(2048)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "_dynet.pyx", line 65, in _dynet.DynetParams.set_mem (_dynet.cpp:5070)
    cpdef set_mem(self, unsigned mem):
  File "_dynet.pyx", line 66, in _dynet.DynetParams.set_mem (_dynet.cpp:5011)
    self.cparams.mem_descriptor = str(mem)
  File "stringsource", line 15, in string.from_py.__pyx_convert_string_from_py_std__in_string (_dynet.cpp:53660)
TypeError: expected bytes, str found
```

All of the other parameter setters described in the documentation work fine, it seems to only be `set_mem` that has an issue. My suspicion is that this may be a Python 3 issue.

I'm using Python 3.5.1 on macOS 10.12.3 with a DyNet install built from source pulled from the GitHub repository today.

**Comment by pmichel31415** [comment]
Yes this is linked to the python version, I will try to fix it

**Comment by pmichel31415** [comment]
Fixed by #360 , you can checkout the branch ` py3_fix_mem` until it's merged

**Comment by benmullins** [comment]
Thank you!
